lib/chat/ui/widgets/chat.dart

```diff
@@ -0,0 +1,97 @@
+import 'package:flutter/material.dart';
+import 'package:farmsmart_flutter/chat/bloc/provider/ChatProvider.dart';
+import 'package:farmsmart_flutter/chat/ui/widgets/bubble_message.dart';
+import 'package:farmsmart_flutter/chat/ui/widgets/chat_list.dart';
+
+class ChatStyle {
+  final Color mainContainerColor;
+  final CrossAxisAlignment crossAxisAlignment;
+
+  const ChatStyle({
+    this.mainContainerColor,
+    this.crossAxisAlignment,
+  });
+
+  ChatStyle copyWith({
+    Color mainContainerColor,
+    CrossAxisAlignment crossAxisAlignment,
+  }) {
+    return ChatStyle(
+      mainContainerColor: mainContainerColor ?? this.mainContainerColor,
+      crossAxisAlignment: crossAxisAlignment ?? this.crossAxisAlignment,
+    );
+  }
+}
+
+class _DefaultStyle extends ChatStyle {
+  final Color mainContainerColor = const Color(0xFFFFFFFF);
+  final CrossAxisAlignment crossAxisAlignment = CrossAxisAlignment.end;
+
+  const _DefaultStyle({
+    Color mainContainerColor,
+    CrossAxisAlignment crossAxisAlignment,
+  });
+}
+
+const ChatStyle _defaultStyle = const _DefaultStyle();
+
+class Chat extends StatelessWidget {
+  final ChatProvider _chatProvider;
+  final ChatStyle _style;
+
+  Chat({
+    @required ChatProvider chatProvider,
+    ChatStyle style,
```

Comment:
Add the defaultStyle here instead of adding below. You won't need to check if the incoming style is null or not

Reply:
Changed